proto/timestamp.go

```diff
@@ -4,6 +4,31 @@ import "time"
 
 // Unix returns the local time.Time corresponding to the given Unix time
 // since January 1, 1970 UTC.
-func (m Timestamp) Unix() time.Time {
```

Comment:
Why it's required to change `this` from value to pointer?

Reply:
to be consistent and safely call Unix() on nil receiver

Reply:
Consistent with what? For example in `std` `func (t Time) Unix() int64`. Still did not get the idea.

> safely call Unix() on nil receiver

I still doubt that it's the best practice. For example from [this](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type):

> If the receiver is a small array or struct that is naturally a value type (for instance, something like the time.Time type), with no mutable fields and no pointers, or is just a simple basic type such as int or string, a value receiver makes sense.
A value receiver can reduce the amount of garbage that can be generated; if a value is passed to a value method, an on-stack copy can be used instead of allocating on the heap. (The compiler tries to be smart about avoiding this allocation, but it can't always succeed.) Don't choose a value receiver type for this reason without profiling first.